### PR TITLE
(fix) O3-2197: Add the ability to remove a patient from a list

### PR DESCRIPTION
**Pull request by jwnasambu**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
 
In this PR, I've added the ability to remove a patient from a list. I've added an icon button to each row in the list members datatable on the list details page. When this button is clicked, a modal prompts the user to confirm that they want to remove the specified patient from the list. Confirming removes the patient from the list and updates the list accordingly.

## Screenshots

https://github.com/openmrs/openmrs-esm-patient-management/assets/33891016/a6a9922c-429a-4250-b563-d92409cd0878

## Related Issue

https://issues.openmrs.org/browse/O3-2197

## Other

*None*

### CODE REVIEW

**Comment by jwnasambu** [comment]
@denniskigen, @ibacher, @vasharma05, @dkayiwa Kindly in reference to this [link](https://github.com/openmrs/openmrs-esm-patient-management/pull/757),  I kindly request you review my work at your convinient time please!

File: packages/esm-patient-list-app/src/api/api-remote.ts [hunk 1]
```diff
@@ -136,6 +146,17 @@ export async function addPatientToList(data: AddPatientData, ac = new AbortContr
   return postData(`${cohortUrl}/cohortmember`, data, ac);
 }
 
+export async function removePatientFromList(cohortMembershipUuid: string) {
+  const ac = new AbortController();
+  return postData(
+    `${cohortUrl}/cohortmember/${cohortMembershipUuid}`,
+    {
+      endDate: new Date(),
```
Comment (denniskigen) [inline packages/esm-patient-list-app/src/api/api-remote.ts:154]: @dkayiwa what should happen after removing a patient from a list using this function? Should subsequently fetching the list return a response whose `endDate` property is set to the value passed above? And further, should the frontend filter the returned cohort data to remove entries with an `endDate` property?

Comment (dkayiwa) [reply packages/esm-patient-list-app/src/api/api-remote.ts:154]: After removing a patient from the list, the original REST call which was used, should return a list which does not include the removed patient.

**Review by denniskigen — APPROVED** [review]

File: packages/esm-patient-list-app/src/api/hooks.ts [hunk 2]
```diff
@@ -134,11 +134,23 @@ export function usePatientListMembers(
...
   v: string = 'full',
 ) {
-  const swrResult = useSWR<FetchResponse<CohortResponse<OpenmrsCohortMember>>, Error>(
+  const { data, error, isLoading, mutate } = useSWR<FetchResponse<CohortResponse<OpenmrsCohortMember>>, Error>(
     `${cohortUrl}/cohortmember?cohort=${patientListUuid}&startIndex=${startIndex}&limit=${pageSize}&v=${v}&q=${searchQuery}`,
     openmrsFetch,
   );
-  return { ...swrResult, data: swrResult?.data?.data?.results };
+
+  // FIXME: This is a workaround for removing a patient from a list
```
Comment (denniskigen) [inline packages/esm-patient-list-app/src/api/hooks.ts:142]: @dkayiwa I've ended up filtering the lists on the frontend to remove patients from the list if a cohort membership has an `endDate` property.

**Pull request merged**